packages/consumption/src/modules/requests/itemProcessors/utility/validateAttributeMatchesWithQuery.ts

```diff
@@ -0,0 +1,198 @@
+        if (result.isError()) return result;
+    } else if (query instanceof RelationshipAttributeQuery) {
+        const result = validateAttributeMatchesWithRelationshipAttributeQuery(query, attribute, recipient);
+        if (result.isError()) return result;
+    } else if (query instanceof ThirdPartyRelationshipAttributeQuery) {
+        const result = validateAttributeMatchesWithThirdPartyRelationshipAttributeQuery(query, attribute, recipient, sender);
+        if (result.isError()) return result;
+    } else {
+        return ValidationResult.error(CoreErrors.requests.unexpectedErrorDuringRequestItemProcessing("An unknown error occurred during the RequestItem processing."));
+    }
+
+    if (query instanceof IdentityAttributeQuery || query instanceof RelationshipAttributeQuery || query instanceof ThirdPartyRelationshipAttributeQuery) {
+        if ((!query.validFrom && attribute.validFrom !== undefined) || (query.validFrom && attribute.validFrom && query.validFrom.isBefore(attribute.validFrom))) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided Attribute is not valid in the queried time frame."));
+        }
+
+        if ((!query.validTo && attribute.validTo !== undefined) || (query.validTo && attribute.validTo && query.validTo.isAfter(attribute.validTo))) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided Attribute is not valid in the queried time frame."));
+        }
+    }
+
+    return ValidationResult.success();
+}
+
+function validateAttributeMatchesWithIdentityAttributeQuery(
+    query: IdentityAttributeQuery,
+    attribute: IdentityAttribute | RelationshipAttribute,
+    recipient: CoreAddress
+): ValidationResult {
+    if (!(attribute instanceof IdentityAttribute)) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided Attribute is not an IdentityAttribute, but an IdentityAttribute was queried."));
+    }
+
+    const recipientIsAttributeOwner = recipient.equals(attribute.owner);
+
+    if (!recipientIsAttributeOwner) {
+        return ValidationResult.error(
+            CoreErrors.requests.attributeQueryMismatch("The provided IdentityAttribute belongs to someone else. You can only share own IdentityAttributes.")
+        );
+    }
+
+    if (query.valueType !== attribute.value.constructor.name) {
+        return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The provided IdentityAttribute is not of the queried IdentityAttribute Value Type."));
+    }
+
+    if (query.tags !== undefined && query.tags.length !== 0) {
+        if (attribute.tags === undefined || attribute.tags.length === 0 || !query.tags.some((aQueriedTag) => attribute.tags!.includes(aQueriedTag))) {
+            return ValidationResult.error(CoreErrors.requests.attributeQueryMismatch("The tags of the provided IdentityAttribute do not contain at least one queried tag."));
+        }
```

Comment:
I once discussed this with someone, but we could discuss again which check could be useful here. Any suggestions?

Reply:
There is another place somewhere in our code where the `tags` are already handled in the way I have currently implemented it. In other words, the `tags` are not validated super strongly currently, but a little. This means that if the Sender does not query any `tags`, the Recipient can add any `tags` to the IdentityAttribute it wants to return. However, if the Sender queries `tags`, the Recipient should return an IdentityAttribute so that at least one of these queried `tags` is used. However, some other can also be added. With this weak validation, I don't think the Recipient's freedom is really restricted, but the Sender's control over the requested IdentityAttributes is ensured to a minimum extent.

Reply:
I don't think the example of the `GetAttributesUseCase` can simply be transferred to this case without giving it a second thought. If I query my own Attributes using a tag, I indeed only expect to find matches. But if I use an optional tag to query Attributes of someone else, I think that is a different discussion.

I'm not rooting too strongly for any side here, still I think it is the correct way to have this discussion now. Otherwise, changing it later could lead to unexpected behavior for users, who are used to it working differently. How about we address this in the next sync?